src/java/org/apache/cassandra/net/InboundMessageHandler.java

```diff
@@ -692,6 +692,12 @@ String id()
         return SocketFactory.channelId(peer, self, type, channel.id().asShortText());
     }
 
+    @Override
+    public String toString()
+    {
+        return id();
+    }
```

Comment:
Assuming there isn't something that actually relied on the default `Object#toString()`, this seemed like a reasonable way to solve the construction problem here. We get the laziness we want, and we get it from `this`, the already created enclosing instance.

Reply:
smart idea